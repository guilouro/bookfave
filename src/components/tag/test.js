import React from 'react'
import Tag from '.'
import { render, fireEvent } from 'react-testing-library'

describe('Tag', () => {
  it('Should render correctly', () => {
    const { container, queryByText } = render(
      <Tag name="wallet" onClose={jest.fn()} />
    )

    expect(queryByText(/wallet/g)).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it('Should call onClose when clicked', () => {
    const onClose = jest.fn()
    const { container } = render(<Tag name="wallet" onClose={onClose} />)

    fireEvent.click(container.querySelector('button'))
    expect(onClose).toBeCalled()
  })
})
