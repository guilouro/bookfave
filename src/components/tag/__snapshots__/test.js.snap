// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Should render correctly 1`] = `
<div>
  <span
    class="sc-bdVaJa gHgWio"
  >
    wallet
    <button
      class="sc-bwzfXH sc-htpNat chgZfo"
    >
      âœ•
    </button>
  </span>
</div>
`;
